runtime/src/server/middleware/get_page_handler.ts

```diff
@@ -109,13 +109,14 @@ export function get_page_handler(
 				preload_error = { statusCode, message };
 			},
 			fetch: (url: string, opts?: any) => {
-				const parsed = new URL.URL(url, `http://127.0.0.1:${process.env.PORT}${req.baseUrl ? req.baseUrl + '/' :''}`);
+				const protocol = req.socket.encrypted ? 'https' : 'http'; // Let's not assume server is listening via http
```

Comment:
I would remove the comment here. it doesn't add anything that's not obvious from the code